# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1744057696 -7200
# Parent  cb0d1fefcb3d8a25d76fbfb3c4b6b9ac227fc661
Feature: RegExp in search terms: Build ICU's regex.

diff --git a/config/external/icu/defs.mozbuild b/config/external/icu/defs.mozbuild
--- a/config/external/icu/defs.mozbuild
+++ b/config/external/icu/defs.mozbuild
@@ -10,17 +10,17 @@
 DEFINES["U_USING_ICU_NAMESPACE"] = 0
 # Don't include obsolete header files.
 DEFINES["U_NO_DEFAULT_INCLUDE_UTF_HEADERS"] = 1
 DEFINES["U_HIDE_OBSOLETE_UTF_OLD_H"] = 1
 
 # Remove chunks of the library that we don't need (yet).
 DEFINES["UCONFIG_NO_LEGACY_CONVERSION"] = True
 DEFINES["UCONFIG_NO_TRANSLITERATION"] = True
-DEFINES["UCONFIG_NO_REGULAR_EXPRESSIONS"] = True
+# DEFINES["UCONFIG_NO_REGULAR_EXPRESSIONS"] = True
 DEFINES["UCONFIG_NO_BREAK_ITERATION"] = True
 DEFINES["UCONFIG_NO_IDNA"] = True
 DEFINES["UCONFIG_NO_MF2"] = True
 
 # We don't need to pass data to and from legacy char* APIs.
 DEFINES["U_CHARSET_IS_UTF8"] = True
 
 # Add 'explicit' keyword to UnicodeString constructors.
diff --git a/config/external/icu/i18n/sources.mozbuild b/config/external/icu/i18n/sources.mozbuild
--- a/config/external/icu/i18n/sources.mozbuild
+++ b/config/external/icu/i18n/sources.mozbuild
@@ -129,19 +129,25 @@ sources = [
     "/intl/icu/source/i18n/olsontz.cpp",
     "/intl/icu/source/i18n/persncal.cpp",
     "/intl/icu/source/i18n/pluralranges.cpp",
     "/intl/icu/source/i18n/plurfmt.cpp",
     "/intl/icu/source/i18n/plurrule.cpp",
     "/intl/icu/source/i18n/quantityformatter.cpp",
     "/intl/icu/source/i18n/rbnf.cpp",
     "/intl/icu/source/i18n/rbtz.cpp",
+    "/intl/icu/source/i18n/regexcmp.cpp",
+    "/intl/icu/source/i18n/regeximp.cpp",
+    "/intl/icu/source/i18n/regexst.cpp",
+    "/intl/icu/source/i18n/regextxt.cpp",
     "/intl/icu/source/i18n/region.cpp",
     "/intl/icu/source/i18n/reldatefmt.cpp",
     "/intl/icu/source/i18n/reldtfmt.cpp",
+    "/intl/icu/source/i18n/rematch.cpp",
+    "/intl/icu/source/i18n/repattrn.cpp",
     "/intl/icu/source/i18n/rulebasedcollator.cpp",
     "/intl/icu/source/i18n/scriptset.cpp",
     "/intl/icu/source/i18n/search.cpp",
     "/intl/icu/source/i18n/selfmt.cpp",
     "/intl/icu/source/i18n/sharedbreakiterator.cpp",
     "/intl/icu/source/i18n/simpletz.cpp",
     "/intl/icu/source/i18n/smpdtfmt.cpp",
     "/intl/icu/source/i18n/smpdtfst.cpp",
@@ -222,23 +228,17 @@ other_sources = [
     "/intl/icu/source/i18n/nortrans.cpp",
     "/intl/icu/source/i18n/nultrans.cpp",
     "/intl/icu/source/i18n/quant.cpp",
     "/intl/icu/source/i18n/rbt.cpp",
     "/intl/icu/source/i18n/rbt_data.cpp",
     "/intl/icu/source/i18n/rbt_pars.cpp",
     "/intl/icu/source/i18n/rbt_rule.cpp",
     "/intl/icu/source/i18n/rbt_set.cpp",
-    "/intl/icu/source/i18n/regexcmp.cpp",
-    "/intl/icu/source/i18n/regeximp.cpp",
-    "/intl/icu/source/i18n/regexst.cpp",
-    "/intl/icu/source/i18n/regextxt.cpp",
-    "/intl/icu/source/i18n/rematch.cpp",
     "/intl/icu/source/i18n/remtrans.cpp",
-    "/intl/icu/source/i18n/repattrn.cpp",
     "/intl/icu/source/i18n/scientificnumberformatter.cpp",
     "/intl/icu/source/i18n/strmatch.cpp",
     "/intl/icu/source/i18n/strrepl.cpp",
     "/intl/icu/source/i18n/titletrn.cpp",
     "/intl/icu/source/i18n/tolowtrn.cpp",
     "/intl/icu/source/i18n/toupptrn.cpp",
     "/intl/icu/source/i18n/translit.cpp",
     "/intl/icu/source/i18n/transreg.cpp",
